rs/consensus/idkg/src/pre_signer.rs

```diff
@@ -167,93 +167,92 @@ impl IDkgPreSignerImpl {
 
-            // Disable the height check on target subnet side for the initial transcripts.
-            // Since the transcript_id.source_height is from the source subnet, the height
-            // cannot be relied upon. This also lets us process the shares for the initial
-            // bootstrap with higher urgency, without deferring it.
-            let msg_height = if target_subnet_xnet_transcripts.contains(&dealing.transcript_id) {
-                None
-            } else {
-                Some(dealing.transcript_id.source_height())
-            };
+        let results = self.thread_pool.install(|| {
+            dealings.into_par_iter().filter_map(|(id, signed_dealing)| {
+                let dealing = signed_dealing.idkg_dealing();
 
-            match Action::action(
-                block_reader,
-                &transcript_param_map,
-                msg_height,
-                &dealing.transcript_id,
-            ) {
-                Action::Process(transcript_params_ref) => {
-                    let transcript_params = match self.resolve_ref(
-                        transcript_params_ref,
-                        block_reader,
-                        "validate_dealings",
-                    ) {
-                        Some(transcript_params) => transcript_params,
-                        None => {
-                            ret.push(IDkgChangeAction::HandleInvalid(
+                // Disable the height check on target subnet side for the initial transcripts.
+                // Since the transcript_id.source_height is from the source subnet, the height
+                // cannot be relied upon. This also lets us process the shares for the initial
+                // bootstrap with higher urgency, without deferring it.
+                let msg_height = if target_subnet_xnet_transcripts.contains(&dealing.transcript_id) {
+                    None
+                } else {
+                    Some(dealing.transcript_id.source_height())
+                };
+
+                match Action::action(
+                    block_reader,
+                    &transcript_param_map,
+                    msg_height,
+                    &dealing.transcript_id,
+                ) {
+                    Action::Process(transcript_params_ref) => {
+                        let dealer_id = signed_dealing.dealer_id();
+                        if !transcript_params_ref
+                            .dealers
+                            .contains(&dealer_id)
+                        {
+                            // The node is not in the dealer list for this transcript
+                            self.metrics.pre_sign_errors_inc("unexpected_dealing");
+                            return Some(IDkgChangeAction::HandleInvalid(
+                                id,
+                                format!("Dealing from unexpected node: {signed_dealing}"),
+                            ))
```

Comment:
```suggestion
                            ));
```
nit, following the same style as new line 229